tests/integration-tests/processors/processors-test.js

```diff
@@ -325,31 +325,48 @@ describe('Processors api', function() {
                 const processorId = uuid();
                 const processor = generateRawJSProcessor('not stored in db processor');
                 const updateResponse = await processorRequestSender.updateProcessor(processorId, processor);
                 should(updateResponse.statusCode).equal(404);
             });
         });
-        describe('DELETE /processors/{processor_id}', function() {
-            it('should return 409 for deleting a processor which is used by other tests', async function() {
-                const processor = generateRawJSProcessor('simple-processor');
+        describe('DELETE /processors/{processor_id}', function () {
+            let test;
+            let testId;
+            let processorId;
+            before(async () => {
+                const processor = generateRawJSProcessor('simple-processor:' + uuid());
                 const processorInsertResponse = await processorRequestSender.createProcessor(processor, validHeaders);
                 should(processorInsertResponse.statusCode).equal(201);
 
-                const processorId = processorInsertResponse.body.id;
-                const test = Object.assign({}, basicTest);
+                processorId = processorInsertResponse.body.id;
+                test = Object.assign({}, basicTest);
                 test.processor_id = processorId;
                 const testsInsertResponse = await testsRequestSender.createTest(test, validHeaders);
                 should(testsInsertResponse.statusCode).equal(201);
+                testId = testsInsertResponse.body.id;
+            });
 
-                const deleteProcessorResponse = await processorRequestSender.deleteProcessor(processorId);
+            it('Should return 409 when trying to delete processor which is in use', async () => {
+                let deleteProcessorResponse = await processorRequestSender.deleteProcessor(processorId);
                 should(deleteProcessorResponse.statusCode).equal(409);
                 should(deleteProcessorResponse.body.message).startWith(`${ERROR_MESSAGES.PROCESSOR_DELETION_FORBIDDEN}: ${test.name}`);
+            });
+
+            it('Should return 409 and name of the correct test name which is using the processor', async function () {
+                test.name = 'new test name';
+                await testsRequestSender.updateTest(test, validHeaders, testId);
+                const deleteProcessorResponse = await processorRequestSender.deleteProcessor(processorId);
+                should(deleteProcessorResponse.statusCode).equal(409);
+                should(deleteProcessorResponse.body.message).startWith(`${ERROR_MESSAGES.PROCESSOR_DELETION_FORBIDDEN}: new test name`);
+            });
+
+            it('Should success deleting the processor after test is deleted', async function () {
+                delete test.processor_id;
+                await testsRequestSender.updateTest(test, validHeaders, testId);
```

Comment:
add another integration test that you implement `Should success deleting the processor after test is deleted`